server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java

```diff
@@ -124,6 +144,14 @@ protected void masterOperation(CreateSubscriptionRequest request,
             );
     }
 
+    private static void checkVersionCompatibility(Version subscriberMinNodeVersion, Settings settings) {
+        Version publishedTableVersion = settings.getAsVersion(IndexMetadata.SETTING_VERSION_CREATED, null);
+        assert publishedTableVersion != null : "All published tables must have version created setting";
+        if (subscriberMinNodeVersion.before(publishedTableVersion)) {
```

Comment:
Please add a test verifying that subscribing to a higher hotfix version works.

```suggestion
        if (subscriberMinNodeVersion.beforeMajorMinor(publishedTableVersion)) {
```

Reply:
Thanks added in https://github.com/crate/crate/pull/14869/commits/04832b31185d380b35cc40d37c460d9c32aa4242